python/pyspark/mllib/tests.py

```diff
@@ -156,6 +156,13 @@ def test_matrix_indexing(self):
             for j in range(2):
                 self.assertEquals(mat[i, j], expected[i][j])
 
+    def test_repr_dense_matrix(self):
+        mat = DenseMatrix(3, 2, [0, 1, 4, 6, 8, 10])
+        self.assertTrue(mat, eval(repr(mat)))
```

Comment:
Using `eval` here is not proper for unit tests. Having `__repr__` return some string that can be evaluated is not the contract. We are not expecting users to use `eval` anyway. This test should be changed to

~~~python
self.assertTrue(repr(mat), "DenseMatrix(3, 2, [0, 1, 4, 5, 8, 10], True)")
~~~